### PR TITLE
test: check types in auto tests

### PR DESCRIPTION
**Pull request by ChristianMurphy**
##### Checklist

- [x] the [contributor license agreement][] is signed
- [x] commit message follows [commit guidelines][]
- [x] tests are included

##### Description of change

I found there was another tsconfig file in the project, this has now been added to the type checking, and the typescript/eslint warnings raised have been resolved.

[contributor license agreement]: https://www.apereo.org/node/676
[commit guidelines]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: autotest/IntegTester/ps/tsrc/cloud.tsx [hunk 1]
```diff
@@ -14,7 +14,7 @@ import {
 } from "@material-ui/core";
 import { ThemeProvider } from "@material-ui/styles";
 import { ProviderRegistrationResponse } from "oeq-cloudproviders/registration";
-import { createRegistration, UpdateRegistration } from "./registration";
+import { createRegistration } from "./registration";
```
Comment (ChristianMurphy) [inline autotest/IntegTester/ps/tsrc/cloud.tsx:17]: typescript errors on unused import `UpdateRegistration`

File: autotest/IntegTester/ps/tsrc/cloud.tsx [hunk 2]
```diff
@@ -157,7 +157,7 @@ ReactDOM.render(
   <React.Fragment>
     <CssBaseline />
     <ThemeProvider theme={theme}>
-      <CloudProvider query={parse(location.search)} />
+      <CloudProvider query={parse(window.location.search)} />
```
Comment (ChristianMurphy) [inline autotest/IntegTester/ps/tsrc/cloud.tsx:160]: eslint marks `location` as a banned global and prefers `window.location` instead.

File: autotest/IntegTester/ps/tsrc/viewitem.tsx [hunk 3]
```diff
@@ -1,7 +1,6 @@
 import * as ReactDOM from "react-dom";
 import * as React from "react";
 import "babel-polyfill";
-import { parse } from "query-string";
```
Comment (ChristianMurphy) [inline autotest/IntegTester/ps/tsrc/viewitem.tsx:4]: typescript errors on unused import.

File: .travis.yml [hunk 4]
```diff
@@ -78,6 +78,9 @@ jobs:
         - cd Source/Plugins/Core/com.equella.core/js
         - npm install-ci
         - cd -
+        - autotest/IntegTester/ps
+        - npm install-ci
```
Comment (ChristianMurphy) [inline .travis.yml:82]: ensures typings are installed for autotests before type checking

File: package.json [hunk 5]
```diff
@@ -10,7 +10,8 @@
     "check:java": "find . -name '*.java' | xargs google-java-format -n --set-exit-if-changed",
     "check:md": "remark -q -f -u remark-validate-links -u remark-lint-no-dead-urls --ignore-pattern NOTICE.md .",
     "check:ts": "eslint \"Source/Plugins/Core/com.equella.core/js/**/*.{js,ts,tsx}\"",
-    "check:ts-types": "tsc --noEmit --project \"Source/Plugins/Core/com.equella.core/js/tsconfig.json\"",
+    "check:ts-types-source": "tsc --noEmit --project \"Source/Plugins/Core/com.equella.core/js/tsconfig.json\"",
+    "check:ts-types-autotest": "tsc --noEmit --project \"autotest/IntegTester/ps/tsconfig.json\"",
```
Comment (ChristianMurphy) [inline package.json:14]: test both source and autotest folders.

File: .travis.yml [hunk 6]
```diff
@@ -78,6 +78,9 @@ jobs:
         - cd Source/Plugins/Core/com.equella.core/js
         - npm install-ci
         - cd -
+        - autotest/IntegTester/ps
```
Comment (edalex-ian) [inline .travis.yml:81]:
```suggestion
        - cd autotest/IntegTester/ps
```

Comment (ChristianMurphy) [reply .travis.yml]: good catch, applied

**Comment by ChristianMurphy** [comment]
> Ah yes, the good ol' IntegTester. Interesting how it sits in a ps folder - I think it's mostly purescript

It is, I didn't realize it was hiding there until #1611 and #1612 opened, and I noticed it was changing `@types` packages in the folder.

**Pull request merged**